syntax = "proto3";

package user_urls.v1;

import "google/protobuf/timestamp.proto";

service UserUrlsService {
    rpc GetUserUrls (GetUserUrlsRequest) returns (GetUserUrlsResponse);
}

message GetUserUrlsRequest {
    int32 limit = 1;
    int32 last_id = 2;
}

message UserUrl {
    string key = 1;
    string target_url = 2;
    string name = 3;
    int32 clicks_count = 4;
    bool is_active = 5;
    google.protobuf.Timestamp created_at = 6;
    google.protobuf.Timestamp last_used = 7;
}

message GetUserUrlsResponse {
    repeated UserUrl items = 1;
    int32 count = 2;
}
